python/Vira/vira_api.py

```diff
@@ -202,16 +213,21 @@ def connect(self, server):
                 server = 'https://' + server
                 vim.command('let g:vira_serv = "' + server + '"')
 
+            option_kwargs.update({
+                'server': server,
+                'verify': cert_verify,
+            })
+
             # Authorize
-            self.jira = JIRA(
-                options={
-                    'server': server,
-                    'verify': cert_verify,
-                },
-                basic_auth=(username, password),
-                timeout=2,
-                async_=True,
-                max_retries=2)
+            kwargs={
+                "timeout": 2,
+                "async_": True,
+                "max_retries": 2,
+                "options": option_kwargs,
+            }
+            kwargs.update(auth_kwargs)
```

Comment:
I kind of hate the way way I construct the final kwargs array, but I didn't want to import some deep-dict-merge module for this. So right now it's a bit ratty and probably fails cyclomatic complexity scores with the janky if/else loop earlier.

I don't think it's quite yet shit enough to refactor :P I do think that dictionary.update sucks by default and doesn't do what one expects (sensible recursive merge ala jsonpatch).